### PR TITLE
(DOCSP-26542): Realm Object Schema/App Services Schema Mapping

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Complete the Realm Object Schema <--> App Services Schema mapping reference docs.

Note: this ticket split from DOCSP-19746, though the git branch name still is DOCSP-19746.

### Jira

- https://jira.mongodb.org/browse/DOCSP-26542

### Staged Changes (Requires MongoDB Corp SSO)

- [Sync Data Model Mapping](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-19746/sync/data-model/data-model-map)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by dacharyc — APPROVED** [review]
The information seems fine and I don't spot any copy issues - just have some Qs about organization but those are Qs and don't necessarily need to be blockers.

File: source/sync/data-model/data-model-map.txt [hunk 1]
```diff
@@ -115,6 +105,300 @@ You can configure the following constraints for a given property:
...
+
+   .. tab::
+      :tabid: kotlin
+
+      The Realm Kotlin SDK does not support Embedded Objects yet.
+
+For more information on modeling to-one relationships in an App Services Schema,
+refer to :ref:`Embedded Object Relationships <relationships-embedded-object>`.
+
+Sets
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sync/data-model/data-model-map.txt:303]: they're not presented in the context of relationships. they're just another data type that has to be mapped btwn realm object schema and app services schema

File: source/sync/data-model/data-model-map.txt [hunk 2]
```diff
@@ -115,6 +105,300 @@ You can configure the following constraints for a given property:
...
+      :tabid: kotlin
+
+      :ref:`kotlin-supported-types`
+
+For more information on modeling array properties in an App Services Schema,
+refer to :ref:`BSON Types - Array <schema-type-array>`.
+App Services Schemas support certain constraints that Realm Object Schemas do not,
+such as specifying the minimum and maximum number of items.
+
+Relationships
```
[the remark this answers is not in the document]
Comment (mongodben) [reply source/sync/data-model/data-model-map.txt:197]: like you think hoist up a lvl to H2? (same as `Mappings`)?

[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sync/data-model/data-model-map.txt:197]: Yeah, then maybe move the data types up above relationships?

[the remark this answers is not in the document]
Comment (mongodben) [reply source/sync/data-model/data-model-map.txt]: moved!

**Pull request merged**